### DISCUSSION TITLE
Not possible to specify a CommandOption with "-d"

### DISCUSSION THREAD

**Discussion by famda** [spectreconsole/spectre.console#772] [Q&A]
Hey,

Noticed that is not possible to specify a CommandOption with "-d":

`
            [CommandOption("-d|--document")]
            public string? Document { get; set; }
`
It always return the error message:
Error: Option 'document' is defined but no value has been provided.

**Comment by patriksvensson** [comment]
To allow the option to recieve a value, you must specify it in the option string:

```
[CommandOption("-d|--document <SOMETHINGHERE>")] public string? Document { get; set; }
```

**Reply by famda** [reply]
This is not working. The same error message appears with that exact syntax.
If I use "-f|--file" as template, it's works as expected.

And if use this template:
"-d|--document [document]"
It returns the same error message of #765

**Comment by patriksvensson** [comment]
I've verified that the proposed solution above works.

<img width="449" alt="image" src="https://user-images.githubusercontent.com/357872/159130496-c76f82de-011a-4039-bbcd-001e284768fd.png">

```csharp
using Spectre.Console;
using Spectre.Console.Cli;
using System.Diagnostics.CodeAnalysis;

var app = new CommandApp<MyCommand>();
return app.Run(args);

public sealed class MyCommand : Command<MyCommand.Settings>
{
    public sealed class Settings : CommandSettings
    {
        [CommandOption("-d|--document <SOMETHINGHERE>")] 
        public string? Document { get; set; }
    }

    public override int Execute([NotNull] CommandContext context, [NotNull] Settings settings)
    {
        AnsiConsole.MarkupLine($"Document: [yellow]{settings.Document}[/]");
        return 0;
    }
}
```

**Reply by famda** [reply]
I have a required command argument "Id" and that CommandOption (document). If I don't pass the argument --document it fails.
The document should be optional.

In your case, what happens if you don't pass the document?

**Reply by patriksvensson** [reply]
It will write `Document: `

**Reply by famda** [reply]
No. It's giving the error message I said on my first message.

**Reply by patriksvensson** [reply]
You will need to submit a minimal reproducible example for this because I can't reproduce what you're saying happens.

<img width="456" alt="image" src="https://user-images.githubusercontent.com/357872/159133090-f521ed8a-f788-4755-8580-691ca39cfea0.png">

**Reply by famda** [reply]
No problem. Here is what I did to reproduce.
[ConsoleApp1.zip](https://github.com/spectreconsole/spectre.console/files/8310861/ConsoleApp1.zip)

![image](https://user-images.githubusercontent.com/26621392/159154402-2cb068de-e26e-4b24-be98-1f649c442e4e.png)

If you change the AddStepCommand to use "-f|--file <document>" instead of "-d|--document <document>" it works as expected.

 ![image](https://user-images.githubusercontent.com/26621392/159154512-6f55fc20-5aaf-4470-8d45-d63235106c1c.png)

**Comment by patriksvensson** [comment]
@famda There is no option called `--id` in your application and when you pass `-id` (one dash), it will parse that as you're passing `-i` and `-d`. This is by design. You do have an argument called `<ID>` at position `0` though, so you should just pass the value as-is.

```
myapp.exe add step THEARGUMENTVALUE -d THEDOCUMENTVALUE
```

It is a bug that it tries to parse the options when it should be parsing the value though.

<img width="321" alt="image" src="https://user-images.githubusercontent.com/357872/159154844-c5bc3180-87f2-40a5-abd4-6aaedca9e1f8.png">

**Reply by famda** [reply]
For some reason, it's not making any sense.

![image](https://user-images.githubusercontent.com/26621392/159155526-04cf0714-6c41-43b5-a604-a55133602a03.png)

It works as expected if I use the template with the "file" instead of document.

Why?

**Reply by famda** [reply]
![image](https://user-images.githubusercontent.com/26621392/159155667-fc821b98-51fd-4b46-83f9-034a81381b44.png)

**Reply by patriksvensson** [reply]
Yeah, `<id>` is an argument, not an option so you can not pass it via `-id` or `--id`.

Just set the value directly.

```
add step test
```

I would recommend that you take a look at our documentation: https://spectreconsole.net/cli/
